crates/proto/src/error.rs

```diff
@@ -350,6 +286,159 @@ impl From<io::Error> for ProtoErrorKind {
     }
 }
 
+/// The error type for errors that get returned in the crate
+#[derive(Error, Clone, Debug)]
+#[non_exhaustive]
+pub struct ProtoError {
+    /// Kind of error that occurred
+    pub kind: ProtoErrorKind,
+    /// Backtrace to the source of the error
+    #[cfg(feature = "backtrace")]
+    pub backtrack: Option<ExtBacktrace>,
+}
+
+impl ProtoError {
+    /// Get the kind of the error
+    #[inline]
+    pub fn kind(&self) -> &ProtoErrorKind {
```

Comment:
same question here RE: accessor for a `pub` field.

Reply:
Removed this also.